util/check-build.sh

```diff
@@ -0,0 +1,149 @@
+if docker images | grep -q "nitro-node"; then
+    echo -e "${GREEN}Nitro Docker image is built.${NC}"
+else
+    echo -e "${RED}Nitro Docker image is not built. Build it using: docker build . --tag nitro-node${NC}"
+    exit 1
+fi
+
+# Step 5: Check prerequisites for building binaries
+echo -e "${BLUE}Checking prerequisites for building Nitro's binaries...${NC}"
+if [[ "$OS" == "Linux" ]]; then
+    prerequisites=(git curl build-essential cmake npm golang clang make gotestsum wasm2wat lld-13 python3 yarn)
+else
+    prerequisites=(git curl make cmake npm go gvm golangci-lint wasm2wat clang gotestsum yarn)
+fi
+
+for pkg in "${prerequisites[@]}"; do
+    if command_exists "$pkg"; then
+        # There is no way to check for wabt / llvm directly, since they install multiple tools
+        # So instead, we check for wasm2wat and clang, which are part of wabt and llvm respectively
+        # and if they are installed, we assume wabt / llvm is installed else we ask the user to install wabt / llvm
+        [[ "$pkg" == "wasm2wat" ]] && pkg="wabt"
+        [[ "$pkg" == "clang" ]] && pkg="llvm"
+
+        # Check for specific symbolic links related to wasm-ld on Linux and macOS
+        if [[ "$pkg" == "llvm" ]]; then
+            if [[ "$OS" == "Linux" ]]; then
+                if [ ! -L /usr/local/bin/wasm-ld ]; then
+                    echo -e "${YELLOW}Creating symbolic link for wasm-ld on Linux.${NC}"
+                    sudo ln -s /usr/bin/wasm-ld-13 /usr/local/bin/wasm-ld
+                else
+                    echo -e "${GREEN}Symbolic link for wasm-ld on Linux is already present.${NC}"
+                fi
+            elif [[ "$OS" == "Darwin" ]]; then
+                if [ ! -L /usr/local/bin/wasm-ld ]; then
+                    echo -e "${YELLOW}Creating symbolic link for wasm-ld on macOS.${NC}"
+                    sudo mkdir -p /usr/local/bin
+                    sudo ln -s /opt/homebrew/opt/llvm/bin/wasm-ld /usr/local/bin/wasm-ld
+                else
+                    echo -e "${GREEN}Symbolic link for wasm-ld on macOS is already present.${NC}"
+                fi
+            fi
+        fi
+
+        echo -e "${GREEN}$pkg is installed.${NC}"
+    else
+        [[ "$pkg" == "wasm2wat" ]] && pkg="wabt"
+        [[ "$pkg" == "clang" ]] && pkg="llvm"
+        echo -e "${RED}$pkg is not installed. Please install $pkg. $INSTALLATION_DOCS_URL${NC}"
+        exit 1
+    fi
+done
+
+# Step 6: Check Node.js version
+if command_exists node && node -v | grep -q "v18"; then
+    echo -e "${GREEN}Node.js version 18 is installed.${NC}"
+else
+    echo -e "${RED}Node.js version 18 not installed. $INSTALLATION_DOCS_URL${NC}"
+    exit 1
+fi
+
+# Step 7a: Check Rust version
+if command_exists rustc && rustc --version | grep -q "1.80.1"; then
+    echo -e "${GREEN}Rust version 1.80.1 is installed.${NC}"
+else
+    echo -e "${RED}Rust version 1.80.1 not installed. $INSTALLATION_DOCS_URL${NC}"
+    exit 1
```

Comment:
This is optional, but I think I'd find it more useful if the script could report on multiple problems instead of stopping as soon as it finds something wrong. Currently, the user has to:

1. run chck-build.sh
2. fix the first problem
3. run check-build.sh
4. fix the next problem
5. ...

If we print out all the problems in one run the user can:

1. run check-build.sh
2. Fix all the problems
3. run check-build.sh
4. Get to work!

Reply:
100%, need to use a separate variable to keep track if any errors have occurred, but run all checks before returning

Reply:
Done.